R/ui-caching.R

```diff
@@ -23,9 +23,25 @@ cache_clear <- function(cache_name = NULL, ask = TRUE) {
 #' Remember the past to be quicker in the future
 #'
 #' Caching makes styler faster on repeated styling and is shared across all APIs
-#' (e.g. `style_text()` and Addin).
-#' That means if you style code that already complies to a
-#' style guide and you have previously styled that code, it will be quicker.
+#' (e.g. `style_text()` and Addin). That means if you style code that already
+#' complies to a style guide and you have previously styled that code, it will
+#' be quicker.
+#'
+#' @section Configuring the cache:
+#'
+#' To comply with the CRAN policy, \{styler\} will by default clean up cache files
+#' that are older than 6 days. This implies that you loose the benefit of the cache
+#' for the files not styled in the last 6 days. You can set the R option
+#' `styler.cache_root` explicitly, (e.g. in your `.Rprofile` with
+#' [usethis::edit_r_profile()]) to
+#'
+#' * confirm the default and silence the message with
+#'   `options(styler.cache_root = "styler")`.
+#'
+#' * opt for an indefinitely long-living cache by setting the option to something
+#'   else (we recommend `options(styler.cache_root = "styler-perm")`, which is
+#'   also used by pre-commit).
```

Comment:
How about the following instead?

```r
#' To comply with the CRAN policy, \{styler\} will by default clean up cache files
#' that are older than 6 days. This implies that you loose the benefit of the cache
#' for the files not styled in the last 6 days. 
#'
#' If you want to avoid this, i.e., if you want the cache to last longer, you can use the 
#' R option `styler.cache_root` to opt for an indefinitely long-lived cache by setting it to 
#' `options(styler.cache_root = "styler-perm")`. 
#'
#' If you are happy with the cache being cleared after 6 days, you can confirm the default and 
#' silence this message by setting it instead to `options(styler.cache_root = "styler")`.
#'
#' You can make this change in your `.Rprofile` using [usethis::edit_r_profile()]). 
```

Reply:
Also fine for me. Where do you see the advantage of your version?

Reply:
I think the suggested write-up has a better _flow_: 
there is one paragraph for what to do if you don't accept default, the other paragraph if you do.

Then the tip about `{usethis}` function. No need to mention pre-commit, which most users might not be familiar with.